### ISSUE TITLE
conda env export not including pip packages

### ISSUE THREAD

**Issue by jvalhondo** [conda/conda#5212]
I'm facing the following issue:
When I do like: 
`conda env export -n docker-labs > enviroment.yml`

I end up with a **deprecated warning** as follows:

> DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.

and the **environment.yml** looks like (which is what I want):
```
name: docker-labs
channels:
- anaconda-fusion
- defaults
dependencies:
- openssl=1.0.2k=1
- pip=9.0.1=py36_1
- python=3.6.1=0
- readline=6.2=2
- setuptools=27.2.0=py36_0
- sqlite=3.13.0=0
- tk=8.5.18=0
- wheel=0.29.0=py36_0
- xz=5.2.2=1
- zlib=1.2.8=3
- pip:
  - brain-learn==1.0.1
  - configparser==3.5.0
  - deap==1.0.2
  - decorator==4.0.11
  - django==1.10.6
  - django-extensions==1.7.7
  - djangorestframework==3.6.2
  - future==0.15.2
  - ipython-genutils==0.2.0
  - jsonschema==2.5.1
  - jupyter-core==4.3.0
  - mock==2.0.0
  - nbformat==4.3.0
  - newrelic==2.78.0.57
  - numpy==1.12.1
  - pandas==0.19.2
  - pbr==3.0.0
  - pillow==2.9.0
  - plotly==2.0.8
  - psycopg2==2.7.1
  - python-dateutil==2.6.0
  - pytz==2017.2
  - requests==2.13.0
  - scikit-learn==0.18.1
  - scipy==0.19.0
  - six==1.10.0
  - traitlets==4.3.2
prefix: path_to_env
```

Thus, I get into my **pip.conf** file and add the following:
```
[list]
format=(legacy|columns)
```

Then I execute the same command:
`conda env export -n docker-labs > enviroment.yml`

And this time I don't get prompted with the previous warning, but the outputs does not include the pip packages. As the outcome looks like:
```
name: docker-labs
channels:
- anaconda-fusion
- defaults
dependencies:
- openssl=1.0.2k=1
- pip=9.0.1=py36_1
- python=3.6.1=0
- readline=6.2=2
- setuptools=27.2.0=py36_0
- sqlite=3.13.0=0
- tk=8.5.18=0
- wheel=0.29.0=py36_0
- xz=5.2.2=1
- zlib=1.2.8=3
prefix: path_to_env
```

So, for some reason it is not including pip packages when I follow the **deprecation warning** 

**Note**: I found this other open issue, but I find that providing also an example where it's not just prompting a deprecation warning, but it's also exporting badly the environment may be useful.
[https://github.com/conda/conda/issues/4596](url)

**Note 2**: Bellow you can find the outcome of my conda info:
```
Current conda install:

               platform : osx-64
          conda version : 4.3.16
       conda is private : False
      conda-env version : 4.3.16
    conda-build version : not installed
         python version : 3.6.0.final.0
       requests version : 2.12.4
       root environment : path/anaconda  (writable)
    default environment : path/anaconda/envs/docker-labs
       envs directories : path/anaconda/envs
                          path/.conda/envs
          package cache : path/anaconda/pkgs
                          path/.conda/pkgs
           channel URLs : https://conda.anaconda.org/anaconda-fusion/osx-64
                          https://conda.anaconda.org/anaconda-fusion/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : path/.condarc
           offline mode : False
             user-agent : conda/4.3.16 requests/2.12.4 CPython/3.6.0 Darwin/16.5.0 OSX/10.12.4
                UID:GID : 501:20
```

**Comment by imhuay** [comment]
same problem at:
```
Current conda install:

               platform : win-64
          conda version : 4.3.17
       conda is private : False
      conda-env version : 4.3.17
    conda-build version : not installed
         python version : 3.6.1.final.0
       requests version : 2.13.0
       root environment : D:\prog\work\Anaconda\VirtualEnvs  (writable)
    default environment : D:\prog\work\Anaconda\VirtualEnvs
       envs directories : D:\prog\work\Anaconda\VirtualEnvs\envs
                      C:\Users\admin\AppData\Local\conda\conda\envs
                      C:\Users\admin\.conda\envs
          package cache : D:\prog\work\Anaconda\VirtualEnvs\pkgs
                      C:\Users\admin\AppData\Local\conda\conda\pkgs
           channel URLs : https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/win-64
                      https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/noarch
                      https://repo.continuum.io/pkgs/free/win-64
                      https://repo.continuum.io/pkgs/free/noarch
                      https://repo.continuum.io/pkgs/r/win-64
                      https://repo.continuum.io/pkgs/r/noarch
                      https://repo.continuum.io/pkgs/pro/win-64
                      https://repo.continuum.io/pkgs/pro/noarch
                      https://repo.continuum.io/pkgs/msys2/win-64
                      https://repo.continuum.io/pkgs/msys2/noarch
            config file : C:\Users\admin\.condarc
           offline mode : False
             user-agent : conda/4.3.17 requests/2.13.0 CPython/3.6.1 Windows/10 Windows/10.0.14393
```

**Comment by rolweber** [comment]
What you should have put into `pip.conf` is

    [list]
    format=legacy

The deprecation warning from pip tells you which two options you have, legacy or columns. The conda tool expects the legacy format.

**Comment by kalefranz** [comment]
resolved with #5274

**Comment by jvalhondo** [comment]
Thanks a lot @rolweber 
🔝
